airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -159,6 +161,7 @@ def __init__(
         self.start, self._concurrent_state = self._get_concurrent_state(stream_state)
         self._lookback_window = lookback_window
         self._slice_range = slice_range
+        self._cursor_granularity = cursor_granularity
```

Comment:
nit: We might have (a very small piece of) logic that is duplicated. My understanding is that this value is basically defined as part of the implementation of `AbstractStreamStateConverter.increment`. Is this fair? If so, would there be a way to not duplicate this logic?

Reply:
The usage of `DateTimeStreamStateConverter.increment()` actually might not be duplicated, but an area that I had not originally accounted for. So in that regard you found a bug I need to fix! The two usages of the method:

- `_compare_interval()`: which is interestingly completely unused in our code
- `merge_intervals()`: I need to fix the increment method so the `DateTimeStreamStateConverter` takes in an optional parameter `cursor_granularity`. And we should use that in place of `timedelta(milliseconds=1)`. Otherwise we won't properly merge partitions together since they won't line up correctly. 

I'll adjust the code accordingly and add a test case for this!